File: lista3/zadanie2/src/Pentagon.cpp
#include <cmath>

#include "Figure.hpp"
#include "ClientInput.hpp"
#include "Pentagon.hpp"

Pentagon::Pentagon(ClientInput input) : sideLength(input.NextInput())
{}

double Pentagon::Area()
{
    return 5. / 4. * pow(sideLength, 2.) / tan(36. * M_PI / 180.);
}

double Pentagon::Perimeter() 
{
    return 5. * sideLength;
}

std::string Pentagon::NameYourself()
{
    return "Pięciokąt"; 
}

File: lista3/zadanie1/Pentagon.java
public class Pentagon extends Figure
{
    public Pentagon(ClientInput input)
    {
        sideLength = input.NextInput();
    }

    public double Area()
    {
        return 5. / 4. * Math.pow(sideLength, 2.) / Math.tan(36. * Math.PI / 180.);
    }

    public double Perimeter()
    {
        return 5. * sideLength;
    }

    public String NameYourself()
    {
        return "Pięciokąt"; 
    }

    private double sideLength;
};
